internal/graph/check.go

```diff
@@ -1203,6 +1205,119 @@ func (c *LocalChecker) checkTTUFastPath(ctx context.Context, req *ResolveCheckRe
 	return c.checkMembership(ctx, req, iter, usersetDetails)
 }
 
+func (c *LocalChecker) onlyOneDirectlyRelatedUserType(ctx context.Context, req *ResolveCheckRequest, rewrite *openfgav1.Userset) (bool, string) {
+	typesys, _ := typesystem.TypesystemFromContext(ctx)
+	reqTupleKey := req.GetTupleKey()
+	objectType := tuple.GetType(reqTupleKey.GetObject())
+	directlRelatedUserTypes, _ := typesys.GetDirectlyRelatedUserTypes(objectType, rewrite.GetTupleToUserset().GetTupleset().GetRelation())
+	if len(directlRelatedUserTypes) != 1 {
+		return false, ""
+	}
+	return true, directlRelatedUserTypes[0].GetType()
+}
+
+func tupleKeyExists(ctx context.Context,
+	ds storage.RelationshipTupleReader,
+	typesys *typesystem.TypeSystem,
+	storeID string,
+	tupleKey *openfgav1.TupleKey, opts storage.ReadOptions, reqContext *structpb.Struct) (bool, error) {
+	usersetIter, err := ds.Read(ctx, storeID, tupleKey, opts)
```

Comment:
nit: i don't understand why you are using `Read` to look for what seems to be a fully complete tuple? Why not `ReadUserTuple`?

Reply:
updated to use ReadUserTuple